python/tank/authentication/sso_saml2/core/sso_saml2_core.py

```diff
@@ -564,9 +573,20 @@ def on_authentication_required(self, reply, authenticator):
         :param reply: Qt reply object. Not used.
         :param authenticator: Qt authenticator object.
         """
-        # Setting the user to an empty string tells the QAuthenticator to
-        # negociate the authentication with the user's credentials.
-        authenticator.setUser('')
+        # We take for granted that if we are on Windows, proper NTLM negociation
+        # is possible between the machine and the IdP. For other platforms, we
+        # pop an authentication dialog.
+        if sys.platform != "win32":
+            auth_dialog = UsernamePasswordDialog()
```

Comment:
If the import failed, the class won't be available. Might want to do something like set UsernamePasswordDialog equal to None in the except clause above, and check sys.platform AND whether that's not None before doing this.

Reply:
Makes total sense. Changed my code to reflect your suggestion.